src/microns_to_nwb/tools/ophys/ophys.py

```diff
@@ -0,0 +1,155 @@
+    )
+
+    # Add type of ROIs
+    plane_segmentation.add_column(
+        name="mask_type",
+        description="The classification of mask as soma or artifact.",
+        data=mask_types.astype(str),
+    )
+
+    return plane_segmentation
+
+
+def _get_fluorescence(nwb, fluorescence_name):
+    ophys = check_module(nwb, "ophys")
+
+    if fluorescence_name in ophys.data_interfaces:
+        return ophys.get(fluorescence_name)
+
+    fluorescence = Fluorescence(name=fluorescence_name)
+    ophys.add(fluorescence)
+
+    return fluorescence
+
+
+def add_roi_response_series(scan_key, field_key, nwb, plane_segmentation):
+    frame_times = (nda.FrameTimes & scan_key).fetch1("frame_times")
+
+    traces_for_each_mask = (nda.Fluorescence() & field_key).fetch("trace", order_by="mask_id")
+    continuous_traces = np.vstack(traces_for_each_mask).T
+
+    roi_table_region = plane_segmentation.create_roi_table_region(
+        region=list(range(continuous_traces.shape[1])), description=f"all rois in field {field_key['field']}"
+    )
+
+    roi_response_series = RoiResponseSeries(
+        name=f"RioResponseSeries{field_key['field']}",
+        description=f"The fluorescence traces for field {field_key['field']}",
+        data=H5DataIO(continuous_traces, compression=True),
+        rois=roi_table_region,
+        timestamps=H5DataIO(frame_times, compression=True),
+        unit="n.a.",
+    )
+
+    fluorescence = _get_fluorescence(nwb=nwb, fluorescence_name="Fluorescence")
+    fluorescence.add_roi_response_series(roi_response_series)
+
+
+def add_ophys(scan_key, nwb):
+    device = nwb.create_device(
+        name="Microscope",
+        description="two-photon random access mesoscope",
+    )
+    ophys = nwb.create_processing_module("ophys", "processed 2p data")
+    image_segmentation = ImageSegmentation()
+    ophys.add(image_segmentation)
+    all_field_data = (nda.Field & scan_key).fetch(as_dict=True)
+    for field_data in all_field_data:
+        optical_channel = OpticalChannel(
+            name="OpticalChannel",
+            description="an optical channel",
+            emission_lambda=500.0,
+        )
+        field_x_in_meters = field_data["field_x"] / 1e6
+        field_y_in_meters = field_data["field_y"] / 1e6
+        field_z_in_meters = field_data["field_z"] / 1e6
+        field_width_in_meters = field_data["um_width"] / 1e6
+        field_height_in_meters = field_data["um_height"] / 1e6
+        imaging_plane = nwb.create_imaging_plane(
+            name=f"ImagingPlane{field_data['field']}",
+            optical_channel=optical_channel,
+            imaging_rate=np.nan,
+            description="no description",
+            device=device,
+            excitation_lambda=920.0,
+            indicator="GCaMP6",
+            location="unknown",
+            grid_spacing=[
+                field_width_in_meters / field_data["px_width"],
+                field_height_in_meters / field_data["px_height"],
```

Comment:
This is producing the values `[2.4999999999999998E-6, 2.5]` in the example file. Can you confirm that makes sense? With units in meters? That second value seems way too large

Reply:
It should be fixed now, https://gui-staging.dandiarchive.org/dandiset/206031/draft